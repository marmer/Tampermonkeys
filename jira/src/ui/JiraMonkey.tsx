import React from 'react';


// TODO: marmer 05.09.2019 make the component refres/reload if the "site changes" (the site content gets replaced by jira)
export default (): React.ReactElement => <div>
    Curtent pate title is: {document.title}
</div>
