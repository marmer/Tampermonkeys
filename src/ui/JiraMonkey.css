#JiraMonkeyToggle {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1000;
}

#JiraMonkeyContainer {
    position: relative;
    left: 10px;
}

.selectedJiraMonkey {
    font-weight: bold;
}

table.monkeyTable td, table.monkeyTable th {
    padding-right: 0.5em;
}

.monkeyFlexColumn {
    margin: 0.5em;
    display: flex;
    flex-direction: column;
}

.monkeyFlexContainer {
    display: flex;
}

.monkeyFlexContainer label {
    font-weight: bold;
    align-self: flex-end;
}
