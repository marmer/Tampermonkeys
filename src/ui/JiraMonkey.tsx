import React, {useState} from 'react';
import WorklogSummarizerView from "./WorklogSummarizerView";
import "./JiraMonkey.css"
import TabPannel from "./TabPannel";
// TODO: marmer 05.09.2019 make the component refresh/reload if the "site changes" (the site content gets replaced by jira)
export default (): React.ReactElement => {
    const [state, setState] = useState({
        toolsVisible: false
    });

    return <div>
        <input id="JiraMonkeyToggle" type="checkbox"
               checked={state.toolsVisible}
               onClick={() => setState({toolsVisible: !state.toolsVisible})}
               title="Jira Monkeys"/>
        {state.toolsVisible && <div id="JiraMonkeyContainer">
            <TabPannel>
                {{
                    title: "First Fancy View",
                    pane: <div> yeah ... the big first pane</div>
                }}
                {{
                    title: "Second Fancy View",
                    pane: <div> yeah ... the big second pane</div>
                }}
            </TabPannel>
            <WorklogSummarizerView/>
        </div>}
    </div>
}


// #### update ticket estimations
// fetch("***/rest/api/2/issue/***", {
//     "method": "PUT",
//     "headers": {
//         "content-type": "application/json",
//         "accept": "application/json"
//     },
//     "body": {
//         "fields": {
//             "timetracking": {
//                 "originalEstimate": "2d",
//                 "remainingEstimate": "3h"
//             }
//         }
//     }
// })

// #### create a new worklog
// fetch("https://jira.schuetze.ag/rest/api/2/issue/****/worklog", {
//     "method": "POST",
//     "headers": {
//         "content-type": "application/json",
//         "accept": "application/json"
//     },
//     "body": {
//         "comment": "Somethign really new",
//         "timeSpent": "2d 1h 10m"
//     }
// })


// #### Update/delete worklog
// fetch("https://jira.schuetze.ag/rest/api/2/issue/****/worklog/*****", {
//     "method": "PUT",
//     "headers": {
//         "content-type": "application/json",
//         "accept": "application/json",
//     },
//     "body": {
//         "comment": "MÃ¶p with update",
//         "timeSpent": "2d"
//     }
// })