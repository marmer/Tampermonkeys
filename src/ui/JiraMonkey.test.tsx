import * as reactTest from "@testing-library/react";
import React from "react";
import JiraMonkey from "./JiraMonkey";

jest.mock("./WorklogSummarizerView", () => (): React.ReactNode => <div>WorklogSummarizerViewMock</div>);
jest.mock("./EstimationShiftView", () => (): React.ReactNode => <div>EstimationShiftViewMock</div>);

describe("JiraMonkey", () => {
    beforeEach(() => {
        localStorage.clear();
    });

    it("should be possible to show and hide the different jira monkeys as well as the whole monkey selection", async () => {
        const jiraMonkey = reactTest.render(<JiraMonkey/>);
        expect(jiraMonkey.container.querySelector("#JiraMonkeyContainer")).toBeNull();
        expect(jiraMonkey.container).toMatchSnapshot("initial state");

        reactTest.fireEvent.click(jiraMonkey.getByTitle("Jira Monkeys"));
        expect(jiraMonkey.container.querySelector("#JiraMonkeyContainer")).not.toBeNull();
        expect(jiraMonkey.getByText("Worklog-summarizer")).toBeVisible();
        expect(jiraMonkey.getByText("Estimation-shifter")).toBeVisible();
        expect(jiraMonkey.getByText("Booking-shifter")).toBeVisible();
        expect(jiraMonkey.getByText("Estimation-fixer")).toBeVisible();
        expect(jiraMonkey.container).toMatchSnapshot("on activated toolbar");

        reactTest.fireEvent.click(jiraMonkey.getByText("Worklog-summarizer"));
        expect(jiraMonkey.container).toMatchSnapshot("on activated toolbar with worklog summarizer");

        reactTest.fireEvent.click(jiraMonkey.getByText("Estimation-shifter"));
        expect(jiraMonkey.container).toMatchSnapshot("on activated toolbar with estimation shifter");

        reactTest.fireEvent.click(jiraMonkey.getByText("Booking-shifter"));
        expect(jiraMonkey.container).toMatchSnapshot("on activated toolbar with booking shifter");

        reactTest.fireEvent.click(jiraMonkey.getByText("Estimation-fixer"));
        expect(jiraMonkey.container).toMatchSnapshot("on activated toolbar with estimation fixer");

        reactTest.fireEvent.click(jiraMonkey.getByTitle("Jira Monkeys"));
        expect(jiraMonkey.container).toMatchSnapshot("on inactive toolbar");
    });

    it("should remind whether it's active or not between different renderings", () => {
        const jiraMonkeyOld = reactTest.render(<JiraMonkey/>);
        reactTest.fireEvent.click(jiraMonkeyOld.getByTitle("Jira Monkeys"));
        const oldStateAsString = reactTest.prettyDOM(jiraMonkeyOld.container);
        jiraMonkeyOld.unmount();

        const jiraMonkeyNew = reactTest.render(<JiraMonkey/>);

        expect(oldStateAsString).toBe(reactTest.prettyDOM(jiraMonkeyNew.container));
    });
});
