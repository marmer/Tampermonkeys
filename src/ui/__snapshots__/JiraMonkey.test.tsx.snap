// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: initial state 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on activated toolbar 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
    <header
      id="JiraMonkeyContainer"
    >
      <div
        class="monkeyToggler"
      >
        <button>
          Worklog-summarizer
        </button>
        <button>
          Estimation-shifter
        </button>
        <button>
          Booking-shifter
        </button>
        <button>
          Estimation-fixer
        </button>
      </div>
      <main />
    </header>
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on activated toolbar with booking shifter 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
    <header
      id="JiraMonkeyContainer"
    >
      <div
        class="monkeyToggler"
      >
        <button>
          Worklog-summarizer
        </button>
        <button>
          Estimation-shifter
        </button>
        <button>
          Booking-shifter
        </button>
        <button>
          Estimation-fixer
        </button>
      </div>
      <main>
        <div>
          BookingShiftViewMock
        </div>
      </main>
    </header>
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on activated toolbar with estimation fixer 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
    <header
      id="JiraMonkeyContainer"
    >
      <div
        class="monkeyToggler"
      >
        <button>
          Worklog-summarizer
        </button>
        <button>
          Estimation-shifter
        </button>
        <button>
          Booking-shifter
        </button>
        <button>
          Estimation-fixer
        </button>
      </div>
      <main>
        <div>
          EstimationFixViewMock
        </div>
      </main>
    </header>
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on activated toolbar with estimation shifter 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
    <header
      id="JiraMonkeyContainer"
    >
      <div
        class="monkeyToggler"
      >
        <button>
          Worklog-summarizer
        </button>
        <button>
          Estimation-shifter
        </button>
        <button>
          Booking-shifter
        </button>
        <button>
          Estimation-fixer
        </button>
      </div>
      <main>
        <div>
          EstimationShiftViewMock
        </div>
      </main>
    </header>
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on activated toolbar with worklog summarizer 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
    <header
      id="JiraMonkeyContainer"
    >
      <div
        class="monkeyToggler"
      >
        <button>
          Worklog-summarizer
        </button>
        <button>
          Estimation-shifter
        </button>
        <button>
          Booking-shifter
        </button>
        <button>
          Estimation-fixer
        </button>
      </div>
      <main>
        <div>
          WorklogSummarizerViewMock
        </div>
      </main>
    </header>
  </div>
</div>
`;

exports[`JiraMonkey should be possible to show and hide the different jira monkeys as well as the whole monkey selection: on inactive toolbar 1`] = `
<div>
  <div>
    <input
      id="JiraMonkeyToggle"
      title="Jira Monkeys"
      type="checkbox"
    />
  </div>
</div>
`;
